thrift/types.go

```diff
@@ -118,8 +118,11 @@ func parseBinaryList(vl []interface{}) ([]byte, error) {
 		case int:
 			// YAML uses int for all small values. For large values, it may use int64 or uint64
 			// but those are not valid bool values anyway, so we don't need to check them.
-			if v < 0 || v >= (1<<8) {
-				return nil, fmt.Errorf("failed to parse list of bytes: %v is not a byte", v)
+			// from https://thrift.apache.org/docs/types:
+			// byte: An 8-bit signed integer
+			// 8-bit signed ranges from -128 to 127
+			if boundary := 1 << 7; v < -boundary || v >= boundary {
```

Comment:
though this is strictly the right thing to do for the original thrift specs which are implemented by apache thrift lib (in java, and other langs): https://github.com/apache/thrift. 
however since the issue lies with thriftrw, so if anyone uses bytes serialized by thriftrw which is in the range 0 to 255, it would fail there. Thus, recommend having this check between -128 to 255 to maintain backward compatibility, and since we are not planning to fix it in thriftrw-go. This will keep both apache thrift and thriftrw-go clients happy!

Reply:
yup this makes sense